src/pages/workspaces/workspace/ShareWorkspaceModal/Collaborator.ts

```diff
@@ -0,0 +1,156 @@
+import colors from 'src/libs/colors';
+import { getUser } from 'src/libs/state';
+import * as Utils from 'src/libs/utils';
+import { BaseWorkspace, isAzureWorkspace, WorkspaceAccessLevel } from 'src/libs/workspace-utils';
+import { AccessEntry, WorkspaceAcl } from 'src/pages/workspaces/workspace/WorkspaceAcl';
+
+interface CollaboratorProps {
+  aclItem: AccessEntry; // the item to render
+  acl: WorkspaceAcl; // the entire current acl
+  setAcl: Dispatch<SetStateAction<WorkspaceAcl>>;
+  originalAcl: WorkspaceAcl; // the original acl, to determine new items
+  workspace: BaseWorkspace; // the workspace the acl belongs to
+  lastAddedEmail?: string; // the most recently added email to the list
+}
+
+/**
+ *  Note: the select menu of each Collaborator item NOT focusable via keyboard
+ *  tab will only focus on the remove buttons
+ */
+export const Collaborator = ({ originalAcl, aclItem, acl, setAcl, workspace, lastAddedEmail }: CollaboratorProps) => {
+  const { email, accessLevel, pending } = aclItem;
+  const POAccessLevel = 'PROJECT_OWNER';
+  const disabled = accessLevel === POAccessLevel || email === getUser().email;
+  const isOld = _.find({ email }, originalAcl);
+
+  return div(
+    {
+      role: 'listitem',
+      style: {
+        display: 'flex',
+        alignItems: 'center',
+        borderRadius: 5,
+        padding: '0.5rem 0.75rem',
+        marginBottom: 10,
+        border: isOld ? `1px solid ${colors.dark(0.25)}` : ` 2px dashed ${colors.success(0.5)}`,
+        backgroundColor: isOld ? colors.light(0.2) : colors.success(0.05),
+      },
+    },
+    [
+      div({ style: { flex: 1 } }, [
+        email,
+        pending && div({ style: styles }, ['Pending']),
+        h(AclInput, {
+          'aria-label': `permissions for ${email}`,
+          autoFocus: email === lastAddedEmail,
+          value: aclItem,
+          onChange: (v) => setAcl(_.map((entry) => (entry.email === email ? v : entry), acl)),
+          disabled,
+          maxAccessLevel: workspace.accessLevel,
+          isAzureWorkspace: isAzureWorkspace(workspace),
+        }),
+      ]),
+      !disabled &&
+        h(
+          Link,
+          {
+            tooltip: 'Remove',
+            onClick: () => {
+              const newAcl = _.remove({ email }, acl);
+              setAcl(newAcl);
+            },
+          },
+          [icon('times', { size: 20, style: { marginRight: '0.5rem' } })]
+        ),
+    ]
+  );
+};
+
+const AclSelect = Select as typeof Select<WorkspaceAccessLevel>;
+type AclSelectProps = Omit<
+  SelectProps<WorkspaceAccessLevel, false, { value: WorkspaceAccessLevel; label: string | undefined }>,
+  'value' | 'onChange' | 'options'
+>;
+
+interface AclInputProps extends AclSelectProps {
+  value: AccessEntry;
+  maxAccessLevel: WorkspaceAccessLevel;
+  isAzureWorkspace: boolean | undefined;
+  disabled: boolean | undefined;
+  onChange: (AccessEntry) => void;
+}
+
+export const AclInput = ({
```

Comment:
let's align these signatures with:

```
export const MyComponent: React.FC<MyProps> = (props) => {
  const { prop1, prop2 } = props;
  // ...
}
```

Reply:
I actually prefer that way of going about it anyway: I just assumed the other was preferred because it's more common  in the project.